### PR TITLE
Remove DoNotInvokeDiagnosticDescriptor class

### LINKED ISSUE #3560: Update message for DoNotInvokeDiagnosticDescriptorDescription
We shouldn't change this (or at least not in this PR), the code is taking care of adding the space + wrapping in single quotes when needed: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Roslyn.Diagnostics.Analyzers/Core/DoNotInvokeDiagnosticDescriptor.cs#L64

_Originally posted by @Evangelink in https://github.com/dotnet/roslyn-analyzers/pull/3543_

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #3560

### CODE REVIEW

File: src/Roslyn.Diagnostics.Analyzers/Core/AnalyzerReleases.Shipped.md [hunk 1]
```diff
@@ -7,7 +7,6 @@ RS0001 | RoslyDiagnosticsPerformance | Warning | SpecializedEnumerableCreationAn
 RS0002 | RoslyDiagnosticsPerformance | Warning | SpecializedEnumerableCreationAnalyzer
 RS0005 | RoslyDiagnosticsPerformance | Warning | CodeActionCreateAnalyzer
 RS0006 | RoslyDiagnosticsReliability | Warning | DoNotMixAttributesFromDifferentVersionsOfMEFAnalyzer
-RS0013 | RoslyDiagnosticsPerformance | Disabled | DiagnosticDescriptorAccessAnalyzer
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/AnalyzerReleases.Shipped.md:10]: Please don't remove rules from shipped file. Instead add a new "Remove Rules" table in Unshipped.md

File: src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ internal static class RoslynDiagnosticIds
         // public const string DoNotUseVerbatimCrefsRuleId = "RS0010";                  // Now RS0010 => XmlDocumentationComments.Analyzers.AvoidUsingCrefTagsWithAPrefixAnalyzer
         // public const string CancellationTokenMustBeLastRuleId = "RS0011";            // Now CA1068 => Microsoft.ApiDesignGuidelines.Analyzers.CancellationTokenParametersMustComeLastAnalyzer
         // public const string DoNotCallToImmutableArrayRuleId = "RS0012";              // Now CA2009 => System.Collections.Immutable.Analyzers.DoNotCallToImmutableCollectionOnAnImmutableCollectionValueAnalyzer
-        public const string DoNotAccessDiagnosticDescriptorRuleId = "RS0013";
+        //public const string DoNotAccessDiagnosticDescriptorRuleId = "RS0013";         // Unused
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs:19]:
```suggestion
        //public const string DoNotAccessDiagnosticDescriptorRuleId = "RS0013";         // Removed
```

File: src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticsAnalyzersResources.resx [hunk 3]
```diff
@@ -26,36 +26,36 @@
...
+    Classes that don't support this are serialized and stored with the
     mimetype set.
-    
-    The mimetype is used for serialized objects, and tells the 
-    ResXResourceReader how to depersist the object. This is currently not 
+
+    The mimetype is used for serialized objects, and tells the
+    ResXResourceReader how to depersist the object. This is currently not
     extensible. For a given mimetype the value must be set accordingly:
-
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticsAnalyzersResources.resx:42]: Please revert all the spacing changes in resx files.

File: src/Roslyn.Diagnostics.Analyzers/Core/AnalyzerReleases.Shipped.md (hunk 1 above)
Comment (Evangelink) [reply src/Roslyn.Diagnostics.Analyzers/Core/AnalyzerReleases.Shipped.md:10]: Ok sorry I wasn't sure about the how to.

File: src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticsAnalyzersResources.resx (hunk 3 above)
Comment (Evangelink) [reply src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticsAnalyzersResources.resx:42]: I don't know what happened, pretty sure I did the change through the resx form page but maybe I did the edit through the xml page... I'll revert! Sorry for the multi-pass review.

File: src/Roslyn.Diagnostics.Analyzers/Core/AnalyzerReleases.Unshipped.md [hunk 4]
```diff
@@ -15,9 +15,13 @@ RS0001 | RoslynDiagnosticsPerformance | Warning | RoslyDiagnosticsPerformance |
...
-RS0013 | RoslynDiagnosticsPerformance | Disabled | RoslyDiagnosticsPerformance | Disabled | DiagnosticDescriptorAccessAnalyzer
 RS0019 | RoslynDiagnosticsReliability | Disabled | RoslyDiagnosticsReliability | Disabled | SymbolDeclaredEventAnalyzer
 RS0023 | RoslynDiagnosticsReliability | Warning | RoslyDiagnosticsReliability | Warning | PartsExportedWithMEFv2MustBeMarkedAsSharedAnalyzer
 RS0032 | RoslynDiagnosticsReliability | Disabled | RoslyDiagnosticsReliability | Disabled | TestExportsShouldNotBeDiscoverable
 RS0033 | RoslynDiagnosticsReliability | Warning | RoslyDiagnosticsReliability | Warning | ImportingConstructorShouldBeObsolete
-RS0034 | RoslynDiagnosticsReliability | Warning | RoslyDiagnosticsReliability | Warning | ExportedPartsShouldHaveImportingConstructor
\ No newline at end of file
+RS0034 | RoslynDiagnosticsReliability | Warning | RoslyDiagnosticsReliability | Warning | ExportedPartsShouldHaveImportingConstructor
+
+### Removed Rules
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/AnalyzerReleases.Unshipped.md:24]: Thanks!

**Pull request merged**